Source/core/Number.h

```diff
@@ -1049,15 +1051,243 @@ namespace Core {
+        ~SInt24() = default;
+
+    public:
+        SInt24& operator=(const SInt24& value) = default;
+        SInt24& operator=(SInt24&& value) = default;
+        SInt24& operator=(const int32_t value)
+        {
+            ASSERT(((static_cast<uint32_t>(value) >> 24) == 0) || ((static_cast<uint32_t>(value) >> 24) == 0xFF));
+            _value = (value | (value & 0x800000 ? 0xFF000000 : 0));
+            return (*this);
+        }
+        operator int32_t() const
+        {
+            return (_value);
+        }
+
+        int32_t AsSInt24() const
+        {
+            return (_value & 0xFFFFFF);
+        }
+
+    private:
+        int32_t _value;
+    };
+
+    class UInt24 {
+    public:
+        static constexpr uint8_t SizeOf = 3;
+        static constexpr uint32_t Max = 0xFFFFFF;
+        static constexpr uint32_t Min = 0;
+
+        using InternalType = uint32_t;
+
+        UInt24()
+            : _value(0)
+        {
+        }
+        UInt24(const UInt24& value) = default;
+        UInt24(UInt24&& value) = default;
+        UInt24(const uint32_t value)
+            : _value(value)
+        {
+            ASSERT((value >> 24) == 0);
+        }
+        ~UInt24() = default;
+
+    public:
+        UInt24& operator=(const UInt24& copy) = default;
+        UInt24& operator=(UInt24&& move) = default;
+        UInt24& operator=(const uint32_t value)
+        {
+            ASSERT((value >> 24) == 0);
+            _value = value;
+            return (*this);
+        }
+        operator uint32_t() const
+        {
+            return (_value);
+        }
+
+        uint32_t AsUInt24() const // just to be consistent with SInt24
+        {
+            return (_value & 0xFFFFFF); // in debug assert should already have fired on assignment
+        }
+
+    private:
+        uint32_t _value;
+    };
+
+    template <typename T, typename std::enable_if<std::is_scalar<T>::value, int>::type = 0>
+    static constexpr uint8_t RealSize()
+    {
+        return (sizeof(T));
+    }
+    template <typename T, typename std::enable_if<T::SizeOf != 0, int>::type = 0>
+    static constexpr uint8_t RealSize()
+    {
+        return (T::SizeOf);
+    }
+        
+    template <typename NEW_TYPE, typename ORIGINAL_TYPE>
+    bool check_and_cast(const ORIGINAL_TYPE& input, NEW_TYPE& output)
+    {
+        ASSERT(input <= std::numeric_limits<NEW_TYPE>::max());
+        ASSERT(input >= std::numeric_limits<NEW_TYPE>::min());
+
+        bool success = false;
+
+        if ((input <= std::numeric_limits<NEW_TYPE>::max()) && (input >= std::numeric_limits<NEW_TYPE>::min())) {
+
+            success = true;
+
+            PUSH_WARNING(DISABLE_WARNING_CONVERSION_POSSIBLE_LOSS_OF_DATA)
+
+            output = static_cast<NEW_TYPE>(input);
+
+            POP_WARNING()
+        }
+
+        return success;
+
+    }
+
 } // namespace Core
+} // namespace Thunder
+
+namespace std { // seems to be allowed/mandatory to specialize inside the std namespace
```

Comment:
Hi @MFransen69 : Blackduck is finding a code  match here from lines 1185-1273, to someone else's Template structure numeric_limits.  But it looks like standard setup for various data types SInt24, UInt24.  Please will you confirm that these lines are simply setup that must be done in this manner, and should not be considered to be copied code.

Reply:
Hi @mhughesacn 

Yes, it is indeed logical it finds matching code is it is a specialization of a C++ standard template. It is  the only way it should be done to add this functionality for your own defined type, which SIn24 and UInt24 are, they are defined in Thunder code (above).
So I can confirm these lines need to be done this way and should not be considered copied code